tasks/cockpit-tasks

```diff
@@ -113,14 +113,29 @@ function run_one_task() {
     fi
 }
 
+function run_from_queue() {
+    cd "$XDG_CACHE_HOME"/cockpit-project/bots
+    # Consuming directly from queue, commands have bots/ prepended
+    [ -e bots ] || ln -s . bots
+    if /usr/bin/timeout 12h /bin/sh -c "set -ex; ./run-queue --amqp \"$AMQP_SERVER\""; then
+        work_done=1
+    fi
+}
+
 # on mass deployment, avoid GitHub stampede
 sleep $(shuf -i 0-120 -n 1)
 
 # Perform a couple of tasks or waits, then restart
 for i in $(seq 1 30); do
     work_done=
 
-    update_repo  "$XDG_CACHE_HOME"/cockpit-project/bots             https://github.com/cockpit-project/cockpit
+    update_repo  "$XDG_CACHE_HOME"/cockpit-project/bots             https://github.com/cockpit-project/bots
+
+    if [ -n "$AMQP_SERVER" ]; then
```

Comment:
Would it be better to have an 80% change to run something from the queue; otherwise run one of the tasks?